lib/hammer_cli_katello/host_collection.rb

```diff
@@ -50,6 +50,8 @@ def request_params
       success_message _("Host collection created.")
       failure_message _("Could not create the host collection")
 
+      option "--unlimited-hosts", :flag, "Set hosts max to unlimited"
```

Comment:
```suggestion
      option "--unlimited-hosts", :flag, "Set hosts max to unlimited"
      
      validate_options :before, 'IdResolution' do
        all(:option_unlimited_hosts, :option_max_hosts).rejected
      end
```
We don't want the user to do something like -> 
```
$ hammer host-collection create --organization-id=1 --unlimited-hosts --name=great2 --max-hosts=1
```